client/helpers.go

```diff
@@ -382,3 +391,28 @@ func TagsToMap(tagSlice interface{}) map[string]string {
 	TagsIntoMap(tagSlice, ret)
 	return ret
 }
+
+func ListAndDetailResolver(ctx context.Context, meta schema.ClientMeta, res chan<- interface{}, list ListResolver, details DetailResolver) error {
+	sem := semaphore.NewWeighted(int64(MAX_GOROUTINES))
+	errs, ctx := errgroup.WithContext(ctx)
+	response, err := list(ctx, meta)
```

Comment:
what if the list is paginated?

Reply:
The `list` function is supposed to handle all of the pagination and only return when it is complete

Reply:
but that could create a huge list of data and memory spikes, it worth noting this to the implementor

Reply:
Now there will not be a spike as there is a channel between the list and detail